Add MAO specs documentation

NutthanichN opened the pull request:
MAO specs documentation for v2.1.0 (latest version). The format of the document is taken from [OpenAPI Specification](https://swagger.io/specification/).

MaiMee1 commented:
There were a lot of parts I'd like to change so I changed it and committed it directly instead of commenting. See if it's all OK and if not, you can revert the changes & take what's OK.

Another thing, are datatypes just properties? I saw you described it like properties so I left it like that.

MaiMee1 commented:
Another option of what I did is to not use `Rules Object` and `Entities Object` and directly use `Map[string, Class Description Object]`, `Map[string, Rule Description Object]`, etc . I think that's a better way of doing it, but it requires some more description in the "Fixed Fields" of `MAO Specs Object`. What do you think? @NutthanichN

NutthanichN commented:
> 
> 
> There were a lot of parts I'd like to change so I changed it and committed it directly instead of commenting. See if it's all OK and if not, you can revert the changes & take what's OK.
> 
> Another thing, are datatypes just properties? I saw you described it like properties so I left it like that.

The changes look good for me (there's some typo and I fixed that in the next commit). About the datatype, its fields are similar to the property's fields so I described it as a property.

NutthanichN commented:
> 
> 
> Another option of what I did is to not use `Rules Object` and `Entities Object` and directly use `Map[string, Class Description Object]`, `Map[string, Rule Description Object]`, etc . I think that's a better way of doing it, but it requires some more description in the "Fixed Fields" of `MAO Specs Object`. What do you think? @NutthanichN

I think we can use this approach as an improvement for the future.
